internal/kafka/internal/handlers/validation.go

```diff
@@ -24,6 +24,39 @@ var ValidKafkaClusterNameRegexp = regexp.MustCompile(`^[a-z]([-a-z0-9]*[a-z0-9])
 
 var MaxKafkaNameLength = 32
 
+func ValidateBillingCloudAccountIdAndMarketplace(ctx context.Context, kafkaService *services.KafkaService, kafkaRequestPayload *public.KafkaRequestPayload) handlers.Validate {
+	blank := func(s *string) bool {
+		return s == nil || *s == ""
+	}
+
+	return func() *errors.ServiceError {
+		// both fields are optional
+		if blank(kafkaRequestPayload.BillingCloudAccountId) && blank(kafkaRequestPayload.Marketplace) {
+			return nil
```

Comment:
should it not continue to `AssignInstanceType` if two fields are empty?

Reply:
No, i don't think so. `AssignInstanceType` is not a good name for what the method does: it only returns if the given org has standard or developer quota. It doesn't really assign anything.

Reply:
yes- you are right, but looking further into the code - you can see in the handler, that AssignInstanceType is called twice. Once here: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1059/files#diff-223cf9d875575c2b0cb237848a3d6055a32acf843ff8fce340d22bfbef6ac5faR63 and once in this block you added. do you think this could be somehow unified?